This patch removes unnecessary generation of a tarball of palemoon.
--- pale-moon.org/platform/toolkit/mozapps/installer/packager-uxp.mk
+++ pale-moon/platform/toolkit/mozapps/installer/packager-uxp.mk
@@ -18,6 +18,8 @@
 ifeq (WINNT,$(OS_ARCH))
 	cd $(DIST); $(CYGWIN_WRAPPER) 7z a -t7z -m0=lzma2 -mx=9 -aoa -bb3 $(PKG_BASENAME).7z $(MOZ_PKG_DIR)
 else
-	cd $(DIST); XZ_OPT=-9e $(TAR) cfJv $(PKG_BASENAME).tar.xz $(MOZ_PKG_DIR)
+	cd $(DIST); 
+	# XZ_OPT=-9e $(TAR) cfJv $(PKG_BASENAME).tar.xz $(MOZ_PKG_DIR)
+	touch $(PKG_BASENAME).tar.xz
 endif
 endif
